l10n_it_ricevute_bancarie/models/account/account.py

```diff
@@ -130,13 +134,33 @@ def _compute_is_unsolved(self):
             if len(invoice.unsolved_move_line_ids) != reconciled_unsolved:
                 invoice.is_unsolved = True
 
-    _inherit = "account.invoice"
     unsolved_move_line_ids = fields.Many2many(
         'account.move.line', 'invoice_unsolved_line_rel', 'invoice_id',
         'line_id', 'Past Due Journal Items')
     is_unsolved = fields.Boolean(
         "Is a past due invoice", compute="_compute_is_unsolved", store=True
     )
+    is_riba_payment = fields.Boolean(
+        'Is C/O Payment', related='payment_term_id.riba', default=False
+    )
+    riba_partner_bank_id = fields.Many2one(
+        'res.partner.bank',
+        string='C/O Bank Account',
+        help='Bank Account Number to which the C/O will be debited. '
+        'If not set, first bank in partner will be used.',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+
+    @api.onchange('partner_id', 'payment_term_id')
```

Comment:
Aggiungerei nell'onchange anche il field `type` e `payment_term_id.riba`.

**Dettaglio**: invertire all'interno dell'`if` l'ordine con cui controlli i vari campi conforme con quello specificato nell'onchange, ovvero: `self.partner_id and self.payment_term_id.riba and self.type == 'out_invoice'`
**Nota**: `self.payment_term_id and self.payment_term_id.riba` lo puoi scrivere semplicemente `self.payment_term_id.riba` in quanto si occupa già l'orm di verificare se i vari field sono null (in questo [punto).](https://github.com/odoo/odoo/blob/640fea880f498d13ca6b72bdf83ad318bbabbcfe/odoo/fields.py#L1025)

Reply:
@tafaRU una domanda, una variazione di `payment_term_id.riba` non viene recepita anche utilizzando il solo `payment_term_id`? E' necessario specificare entrambe?

Reply:
Se la valutazione la fai esclusivamente su `payment_term_id.riba` metterei solo questo campo nell'onchange.
Mettendo solo `payment_term_id` una modifica sul suo campo `riba` non penso venga recepita.

Reply:
Nel frattempo ho scoperto che la dot notation non supporta la dot notation: https://github.com/odoo/odoo/blob/0617a4724cfda5dc809473ba24e5ae7e5ce965ec/odoo/api.py#L203-L207

Reply:
@GSLabIt in base a quanto scritto sopra puoi lasciare solo `payment_term_id`.
Ricapitolando: `@api.onchange('partner_id', 'payment_term_id', 'type')`.

Reply:
@tafaRU ok, modifico.